contrib/systemd/system/podman-restart.service.in

```diff
@@ -10,7 +10,7 @@ Type=oneshot
 RemainAfterExit=true
 Environment=LOGGING="--log-level=info"
 ExecStart=@@PODMAN@@ $LOGGING start --all --filter restart-policy=always
-ExecStop=/bin/sh -c '@@PODMAN@@ $LOGGING stop $(@@PODMAN@@ container ls --filter restart-policy=always -q)'
+ExecStop=/bin/sh -c 'CONTAINERS="$(@@PODMAN@@ container ls --filter restart-policy=always -q)"; if [ -n "$CONTAINERS" ]; then @@PODMAN@@ $LOGGING stop $CONTAINERS; fi'
```

Comment:
I have no idea why stop was made special here via the shell script.
podman stop supports the same filter as start AFAICT.

So IMO this really should just be `@@podman@@ stop --filter restart-policy=always` so need need for extra commands at all

Reply:
That seems more intuitively correct, but further from the previous intent, so I've opened #25134 if you'd rather go that way.  Please accept whichever one is most acceptable :)